app/views/lists/_datatable.html.erb

```diff
@@ -43,271 +44,37 @@
-     <% end %>
-   });
-
-   var tbl = $('#datatable-table').DataTable();
-
-   var type = $('#datatable-type');
-   type.on('change', function() {
-     var val = $(this).val() ? "\\b" + $(this).val() + "\\b" : "";
-     tbl.columns('types:name').search(val, true).draw();
-   });
-
-   var industry = $('#datatable-industry');
-   industry.on('change', function() {
-     var val = $(this).val() ? "\\b" + $(this).val() + "\\b" : "";
-     tbl.columns('industries:name').search(val, true).draw();
-   });
-
-   var interlock = $('#datatable-interlock');
-   interlock.on('change', function() {
-     var val = $(this).val() ? "\\b" + $(this).val() + "\\b" : "";
-     tbl.columns('interlock_ids:name').search(val, true).draw();
-   });
-
-   var list_interlock = $('#datatable-list-interlock');
-   list_interlock.on('change', function() {
-     var val = $(this).val() ? "\\b" + $(this).val() + "\\b" : "";
-     tbl.columns('list_interlock_ids:name').search(val, true).draw();
-   });
-
-   var search = $('#datatable-search');
-   search.keyup(function() {
-     tbl.columns('master_search:name').search($(this).val()).draw();
-   });
-
-   var array_to_csv_data = function(ary) {
-     var str = "data:text/csv;charset=utf-8,";
-     var lines = ary.map(function(data){
-       return data.join(",");
-     });
-     str += lines.join("\n");    
-     return encodeURI(str);
-   }
-
-   var escape_csv_value = function(value) {
-     var value = value === null ? '' : value.toString();
-     value = value.replace(/"/g, '""');
-     if (value.search(/("|,|\n)/g) >= 0) {
-       value = '"' + value + '"';      
-     }
-     return value;  
-   }
-
-   var a = document.createElement('a');
-   a.innerHTML = 'Export CSV';
-   a.setAttribute('class', 'btn btn-primary');
-   a.href="#";
-   $('.buttons')[0].appendChild(a);
-   $(a).on('click', function() {
-     var fields = [<%= table.ranked? ? raw("'rank', ") : "" %>'id', 'name', 'blurb', 'types']
-     var data = [fields]
-     var rows = $('#datatable-table').dataTable()._('tr', { filter: "applied" });
-     data = data.concat(Array.prototype.slice.apply(rows).map(function(d) {
-       return fields.map(function(field) {
-         return escape_csv_value(d[field]);
-       });
-     }));
-     window.open(array_to_csv_data(data));
-   });
-
-   var init_search = "";
-   var hash = (window.location.href.split('#').length > 1) ? window.location.href.split('#')[1] : "";
-   hash.split('&').forEach(function(pair) {
-     pair = pair.split('=');
-     var key = pair[0];
-     var val = decodeURIComponent(pair[1]);
-
-     var e = $('#datatable-' + key);
-
-     if (str_fields.indexOf(key) !== -1) {
-       e.val(val);
-       e.trigger('change');
-
-     } 
-
-     if (key == 'search') {
-       init_search = val;
-     }
-   });
-
-   tbl.columns('master_search:name').search(init_search).draw();
- });
+  $(document).ready(function() {
+    let loader = new LittleSis.listDatatableLoader({
```

Comment:
I get a javascript error when loading the page here:  `Uncaught TypeError: LittleSis.listDatatableLoader is not a constructor`

Reply:
Hmmn. Works for me. Which browser are you using?

Reply:
firefox 79. I'll try to do more investigating -- it could be something I changed locally.

Reply:
I've tested against Firefox 76 and 78 (the latest I can get on my distribution), and Chromium 81, and it seems to be OK. Perhaps `yarn` related?